File: Remove Duplicates from Sorted List II.js
var deleteDuplicates = function(head) {
    var current = head;
    var prev = head;
   
    while(current){
        if(!current.next) break;
        if(current.val === current.next.val){
            if(current === head){
                while(current.val === current.next.val){
                    current = current.next;
                    if(!current.next) return null;
                } 
                head = current.next;
            }else{
                while(current.val === current.next.val){
                    current = current.next;
                    if(!current.next) break;
                } 
                prev.next = current.next;
                current = prev;
            } 
        }else{
            prev = current;
            current = current.next;
        } 
    }
    return head;
};

File: Remove Duplicates from Sorted List II.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        behind = dummy
        curr = head
        isduplicated = False

        while curr:
            isduplicated = False
            while curr and curr.next and curr.val == curr.next.val:
                curr = curr.next
                isduplicated = True
            
            if isduplicated:
                behind.next = curr.next
            else:
                behind.next = curr
                behind = behind.next
            curr = curr.next

        return dummy.next
